spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_weight_changable.py

```diff
@@ -0,0 +1,390 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+from typing import Iterable, List, Optional, Tuple, Dict, TYPE_CHECKING, cast
+
+import numpy
+from numpy import floating, integer, uint8, uint16, uint32
+from numpy.typing import NDArray
+
+from pyNN.standardmodels.synapses import StaticSynapse
+
+from spinn_utilities.overrides import overrides
+
+from spinn_front_end_common.interface.ds import DataSpecificationBase
+from spinn_front_end_common.utilities.constants import (
+    BYTES_PER_WORD, BYTES_PER_SHORT)
+
+from spynnaker.pyNN.exceptions import SynapticConfigurationException
+from spynnaker.pyNN.models.neural_projections.connectors import (
+    AbstractConnector)
+from spynnaker.pyNN.types import Weight_Types
+from spynnaker.pyNN.types import Weight_Delay_In_Types as _In_Types
+from spynnaker.pyNN.utilities.utility_calls import get_n_bits
+from spynnaker.pyNN.models.neuron.synapse_dynamics.types import (
+    NUMPY_CONNECTORS_DTYPE)
+from .abstract_plastic_synapse_dynamics import AbstractPlasticSynapseDynamics
+from .abstract_generate_on_machine import (
+    AbstractGenerateOnMachine, MatrixGeneratorID)
+
+if TYPE_CHECKING:
+    from spynnaker.pyNN.models.neural_projections import (
+        ProjectionApplicationEdge, SynapseInformation)
+    from spynnaker.pyNN.models.neuron.synapse_dynamics.types import (
+        ConnectionsArray)
+    from spynnaker.pyNN.models.neuron.synapse_io import MaxRowInfo
+    from .abstract_synapse_dynamics import AbstractSynapseDynamics
+
+# How large are the time-stamps stored with each event
+TIME_STAMP_BYTES = BYTES_PER_WORD
+
+# The targets of neuromodulation
+NEUROMODULATION_TARGETS = {
+    "reward": 0,
+    "punishment": 1
+}
+
+
+class SynapseDynamicsWeightChangable(
+        AbstractPlasticSynapseDynamics, AbstractGenerateOnMachine):
+    """
+    The dynamics of a synapse that can be changed simply by the sending of an
+    external signal.
+    """
+
+    __slots__ = (
+
+        # The maximum weight
+        "__weight_max",
+
+        # The minimum weight
+        "__weight_min",
+
+        # The map of synapse information to index
+        "__synapse_info_to_index",
+
+        # The next index to use for the next projection
+        "__next_index")
+
+    def __init__(
+            self,
+            weight_min: float, weight_max: float,
+            weight: _In_Types = StaticSynapse.default_parameters['weight'],
+            delay: _In_Types = None):
+        """
+        :param float weight:
+        :param delay: Use ``None`` to get the simulator default minimum delay.
+        :type delay: float or None
+        """
+        super().__init__(delay=delay, weight=weight)
+        self.__weight_max = weight_max
+        self.__weight_min = weight_min
```

Comment:
max and min should be validated!
1. Make sure max is not smaller
2. make sure they are values the code can handle